operator/api/v1alpha1/central_types.go

```diff
@@ -475,9 +491,55 @@ type ExposureRoute struct {
 	Enabled *bool `json:"enabled,omitempty"`
 
 	// Specify a custom hostname for the central route.
-	// If unspecified, an appropriate default value will be automatically chosen by OpenShift route operator.
+	// If unspecified, an appropriate default value will be automatically chosen by the OpenShift route operator.
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2
 	Host *string `json:"host,omitempty"`
+
+	// Set up a central route with reencrypt tls termination.
+	// For reencrypt routes, the request is terminated on the OpenShift router with a custom certificate.
+	// The request is then reencrypted by the OpenShift router and sent to central.
+	// <user> --tls--> <OpenShift router> --tls--> <central>
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=3
+	Reencrypt *ExposureRouteReencrypt `json:"reencrypt,omitempty"`
+}
+
+// ExposureRouteReencrypt defines settings for exposing central via a reencrypt Route.
+type ExposureRouteReencrypt struct {
+	//+kubebuilder:default=false
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Specify a custom hostname for the central reencrypt route.
+	// If unspecified, an appropriate default value will be automatically chosen by the OpenShift route operator.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2
+	Host *string `json:"host,omitempty"`
+
+	// TLS settings for exposing central via a reencrypt Route.
+	TLS *ExposureRouteReencryptTLS `json:"tls,omitempty"`
+}
+
+// ExposureRouteReencryptTLS defines TLS settings for exposing central via a reencrypt Route.
+type ExposureRouteReencryptTLS struct {
+	// PEM encoded certificate chain that may be used to establish a complete chain of trust.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	CaCertificate *string `json:"caCertificate,omitempty"`
+
+	// The PEM encoded certificate that is served on the route. Must be a single serving
+	// certificate instead of a certificate chain.
+	// Defaults to a certificate signed by the OpenShift certificate authority is used.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2
+	Certificate *string `json:"certificate,omitempty"`
+
+	// The CA certificate of the final destination. Should be provided because the OpenShift
```

Comment:
Using the default (central CA) sounds sensible, so why should the user provide one? I'm in particular worried that it will complicate CA cert rotation which is being added by @vladbologa as we speak.

Or do you mean that `Should be provided` applies to us as operator authors? In that case please remove this part of the sentence.

Reply:
The user needs to provide the destination CA if they have Central serve a custom certificate. Then the operator cannot know the CA that signed the custom cert, and it must be provided here - otherwise the route won't work. If Central serves a cert from the default stackrox CA, this can be left empty.

Reply:
Removed the `should be provided` part.